app/models/emoji.rb

```diff
@@ -48,19 +43,30 @@ def self.exists?(name)
   end
 
   def self.[](name)
-    Emoji.custom.detect { |e| e.name == name }
+    name = name.delete_prefix(':').delete_suffix(':')
+    is_toned = name.match?(/.+:t[1-6]/)
+    normalized_name = name.gsub(/(.+):t[1-6]/, '\1')
+
+    Emoji.all.detect do |e|
+      e.name == normalized_name &&
+      (!is_toned || is_toned && e.tonable)
+    end
   end
 
   def self.create_from_db_item(emoji)
     name = emoji["name"]
     filename = emoji['filename'] || name
+    is_tonable = Emoji.tonable_emojis.include?(name)
+
     Emoji.new.tap do |e|
       e.name = name
+      e.tonable = is_tonable
```

Comment:
`is_tonable` feels unnecessary here.

```suggestion
      e.tonable = Emoji.tonable_emojis.include?(name)
```